Katello/katello
# Fixes #33951 - New settings for disconnected

#### What are the changes introduced in this pull request?

This PR breaks the `Content Disconnected` setting into 2 parts. 
-  A new setting called `Subscription  Connection Enabled` which indicates whether a Connection to the subscription portal is available. If so then the user is able to connect and update entitlements.
![Settings](https://user-images.githubusercontent.com/1069779/144699414-753c19e3-8ee8-4644-8876-f0d37cb7826e.png)

- A new toggle in the CdnConfiguration whether there is a Connection/Air-gapped. This information would be used to determine if they are able enable/disable redhat repos or if they are purely import.
![Airgapped](https://user-images.githubusercontent.com/1069779/144699416-4bf860cb-9593-4124-adcb-2f99b786d419.png)


![Airgapped-RH-Repos](https://user-images.githubusercontent.com/1069779/144699415-ec558aa8-3bab-4d0a-92e0-293725faa90e.png)

#### Considerations taken when implementing this change?

We are trying to balance 2 different meanings of the `Disconnected`setting and classifying it as `Subscription Portal Connection Enabled` and `Airgapped`. These 2 settings work together as follows
| Subscription Connection Enabled| Airgapped| Implcation |
| --- | --- | --- |
| Yes | No | The Satellite is able to access the portal and fetch connection from the CDN.|
| No | No | The Satellite is not able to access the portal or cdn but repositories can be enabled/disabled and content is fetched from another satellite via direct connection .|
| No | Yes| No portal/cdn access and  but repos cannot be enabled or disabled and have to use to Import/Export for content.|
| Yes | Yes | Highly unlikely condition where you can access portal but for some reason want to force using Import/Export for content.|

#### What are the testing steps for this pull request?

-  Run the migration
- Make sure `Subscription Connection Enabled` points to the opposed of what `Content Disconnected` was.
- Make sure there is  no `Content Disconnected` entry
- Try playing with `Subscription Connection Enabled` 
- Go to `Subscriptions-> Manage Manifest -> CDN configuration`. Play with the `Air-Gapped` toggle
-  Make sure you are able/unable to set your entitlements based on the value of `Subscription Connection Enabled`
- Make sure you are able/unable to enable/disable RH Repos based on the value of Airgapped.
